pandas/io/formats/xml.py

```diff
@@ -0,0 +1,601 @@
+        Adjust Data Frame to fit xml output.
+
+        This method will adjust underlying data frame for xml output,
+        including replacing missing entities and including indexes.
+        """
+
+        na_dict = {"None": self.na_rep, "NaN": self.na_rep, "nan": self.na_rep}
+
+        df = (
+            (self.fmt.frame.reset_index().applymap(str).replace(na_dict))
+            if self.index
+            else self.fmt.frame.applymap(str).replace(na_dict)
+        )
+
+        return df.to_dict(orient="index")
+
+    def handle_indexes(self) -> None:
+        """
+        Handle indexes.
+
+        This method will add indexes into attr_cols or elem_cols.
+        """
+
+        indexes: List[str] = [
+            x for x in self.frame_dicts[0].keys() if x not in self.orig_cols
+        ]
+
+        if self.attr_cols and self.index:
+            self.attr_cols = indexes + self.attr_cols
+
+        if self.elem_cols and self.index:
+            self.elem_cols = indexes + self.elem_cols
+
+    def get_prefix_uri(self) -> str:
+        """
+        Get uri of namespace prefix.
+
+        This method retrieves corresponding URI to prefix in namespaces.
+
+        Raises
+        ------
+        KeyError
+            *If prefix is not included in namespace dict.
+        """
+
+        raise AbstractMethodError(self)
+
+    def other_namespaces(self) -> dict:
+        """
+        Define other namespaces.
+
+        This method will build dictionary of namespaces attributes
+        for root element, conditionally with optional namespaces and
+        prefix.
+        """
+
+        nmsp_dict: Dict[str, str] = {}
+        if self.namespaces and self.prefix is None:
+            nmsp_dict = {"xmlns": n for p, n in self.namespaces.items() if p != ""}
+
+        if self.namespaces and self.prefix:
+            nmsp_dict = {"xmlns": n for p, n in self.namespaces.items() if p == ""}
+
+        return nmsp_dict
+
+    def build_attribs(self) -> None:
+        """
+        Create attributes of row.
+
+        This method adds attributes using attr_cols to row element and
+        works with tuples for multindex or hierarchical columns.
+        """
+
+        raise AbstractMethodError(self)
+
+    def build_elems(self) -> None:
+        """
+        Create child elements of row.
+
+        This method adds child elements using elem_cols to row element and
+        works with tuples for multindex or hierarchical columns.
+        """
+
+        raise AbstractMethodError(self)
+
+    def write_output(self) -> Optional[str]:
+        xml_doc = self.build_tree()
+        out_str: Optional[str] = xml_doc.decode(self.encoding).rstrip()
+
+        try:
+            if self.path_or_buffer and isinstance(self.path_or_buffer, str):
+                with open(self.path_or_buffer, "wb") as f:
+                    f.write(xml_doc)
+
+                out_str = None
+
+        except (UnicodeDecodeError, OSError, FileNotFoundError) as e:
+            raise e
```

Comment:
why are you catching this specifically? the try/except really isn't doing anything

Reply:
Hmmm...actually this does catch exceptions. If user passes in a file name that cannot be saved, `FileNotFoundError` is raised and there is a test for that: `geom_df.to_xml("/my/fake/path/output.xml", parser=parser)`. I am removing `UnicodeDecodeError` which relates more to Python 2 (borrowed from `to_html`). `OSError` is general class error and to test I need permissions issue, disk full error, or other OS reasons.

Reply:
sure but since you *aren't* changing the exception (type or message), this is not doing anything. just remove the try/except.

Reply:
Got it. Will do.